assembly/src/parsers/blocks.rs

```diff
@@ -251,22 +254,45 @@ impl BlockParser {
 // ================================================================================================
 
 pub fn combine_blocks(mut blocks: Vec<CodeBlock>) -> CodeBlock {
-    // merge consecutive Span blocks
+    // merge consecutive Span blocks.
     let mut merged_blocks: Vec<CodeBlock> = Vec::with_capacity(blocks.len());
+    // Keep track of all the consecutive Span blocks and are merged together when there is a discontinuity.
+    let mut continuous_span_vector: Vec<Span> = Vec::new();
+    // flag for last block being a SPAN block or not.
+    let mut is_last_block_span = false;
+
     blocks.drain(0..).for_each(|block| {
         if block.is_span() {
-            if let Some(CodeBlock::Span(last_span)) = merged_blocks.last_mut() {
-                // this is guaranteed to execute because we know that the block is a span
-                if let CodeBlock::Span(span) = block {
-                    last_span.append(span);
```

Comment:
Do we still need `append()` function in the `Span`? If not, we should probably get rid of it.

Reply:
Removed! Please see [e6a4151](https://github.com/maticnetwork/miden/pull/250/commits/e6a4151355e95dd075de309a5c0d125d90772aae).